docs/rules/prefer-result-array-groups.md

```diff
@@ -0,0 +1,63 @@
+---
+pageClass: "rule-details"
+sidebarDepth: 0
+title: "regexp/prefer-result-array-groups"
+description: "enforce using result array `groups`"
+---
+# regexp/prefer-result-array-groups
+
+> enforce using result array `groups`
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
+- :wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.
+
+## :book: Rule Details
+
+This rule reports and fixes index access in regexp result array that do not use the name of their referenced capturing group.
+
+<eslint-code-block fix>
+
+```js
+/* eslint regexp/prefer-result-array-groups: "error" */
+
+const regex = /(?<foo>a)(b)c/
+let re
+while (re = regex.exec(str)) {
+    /* ✓ GOOD */
+    var p1 = re.groups.foo
+    var p2 = re[2]
+
+    /* ✗ BAD */
+    var p1 = re[1]
```

Comment:
I know this is really pedantic, but could we rename `re`? People typically use `re` to mean regular expression but this variable holds a regex match. 
```suggestion
let match
while (match = regex.exec(str)) {
    /* ✓ GOOD */
    var p1 = match.groups.foo
    var p2 = match[2]

    /* ✗ BAD */
    var p1 = match[1]
```